packages/bigcommerce/src/api/endpoints/signup/signup.ts

```diff
@@ -22,28 +21,27 @@ const signup: SignupEndpoint['handlers']['signup'] = async ({
         },
       ]),
     })
+
+    // Login the customer right after creating it
+    const response = await commerce.login({
+      variables: { email, password },
+      config,
+    })
+
+    return response
   } catch (error) {
-    if (error instanceof BigcommerceApiError && error.status === 422) {
-      const hasEmailError = '0.email' in error.data?.errors
-      // If there's an error with the email, it most likely means it's duplicated
-      if (hasEmailError) {
-        throw new CommerceAPIError('Email already in use', {
+    // Display all validation errors from BigCommerce in a single error message
+    if (error instanceof BigcommerceApiError && error.status >= 400) {
+      const message = Object.values(error.data.errors).join('<br />')
+      if (message) {
+        throw new CommerceAPIError(message, {
           status: 400,
-          code: 'duplicated_email',
+          code: 'invalid_request',
         })
       }
-    } else {
-      throw error
```

Comment:
I understand improving the error handling here, but why was the duplicated email check removed?

Reply:
This displays the actual error and if there are multiple errors. And I checked, and the BigCommerce message was more explicit. I modified this because BigComerce might throw many invalid requests/validations that, at signup, showed it a generic 500 error message for a password that was seven characters but not a numbers & letters combination. It still catches the duplicated email.